### ISSUE TITLE
Linter parses third-party content installed outside of the project directory

### ISSUE THREAD

**Issue by btravouillon** [ansible/ansible-lint#2536]
This is a follow-up of https://github.com/ansible/ansible-lint/discussions/2067.

When third-party roles are already installed in the default path (`~/.ansible/`), `ansible-lint` checks this content and report errors and warnings external to the project that one wants to check.

> I know for sure that the linter is not supposed to look outside the current project.

It still does with 6.8.0a0 and 6.8.0b1.

Reproducible in a container where the ansible requirements are already installed in the default path.

## Dockerfile
```
FROM registry.access.redhat.com/ubi8

RUN dnf install python39 git -y

RUN pip3 install -U pip
RUN pip install ansible ansible-lint==6.8.0a0

RUN mkdir -p projectA/playbooks/

RUN echo "roles:" > projectA/requirements.yml && \
    echo "  - name: minio" >> projectA/requirements.yml && \
    echo "    src: https://github.com/minio/ansible-minio" >> projectA/requirements.yml

RUN ansible-galaxy role install -r projectA/requirements.yml

RUN echo "- name: Reproducer" > projectA/playbooks/minio.yml && \
    echo "  hosts: localhost" >> projectA/playbooks/minio.yml && \
    echo "  roles:" >> projectA/playbooks/minio.yml && \
    echo "    - role: minio" >> projectA/playbooks/minio.yml
```

## Reproducer
```
$ docker build -t "ansible-lint:2067" .
$ docker run -ti ansible-lint:2067
[root@b7cb8306777c /]# cd projectA/
[root@b7cb8306777c projectA]# ansible-lint --version
ansible-lint 6.8.0a0 using ansible 2.13.4
[root@b7cb8306777c projectA]# ansible-lint playbooks/
WARNING  Listing 25 violation(s) that are fatal
jinja: Jinja2 spacing could be improved: {{ minio_server_ip }}:{{ minio_server_port}} -> {{ minio_server_ip }}:{{ minio_server_port }} (jinja[spacing])
~/.ansible/roles/minio/defaults/main.yml:30 Jinja2 template rewrite recommendation: `{{ minio_server_ip }}:{{ minio_server_port }}`.

fqcn: Use FQCN for builtin module actions (systemd). (fqcn[action-core])
~/.ansible/roles/minio/handlers/main.yml:4 Use `ansible.builtin.systemd` or `ansible.legacy.systemd` instead.

name: All names should start with an uppercase letter. (name[casing])
~/.ansible/roles/minio/handlers/main.yml:4 Task/Handler: reload minio systemd

yaml: truthy value should be one of [false, true] (yaml[truthy])
~/.ansible/roles/minio/handlers/main.yml:7

fqcn: Use FQCN for builtin module actions (service). (fqcn[action-core])
~/.ansible/roles/minio/handlers/main.yml:9 Use `ansible.builtin.service` or `ansible.legacy.service` instead.

name: All names should start with an uppercase letter. (name[casing])
~/.ansible/roles/minio/handlers/main.yml:9 Task/Handler: restart minio

fqcn: Use FQCN for builtin module actions (uri). (fqcn[action-core])
~/.ansible/roles/minio/handlers/main.yml:15 Use `ansible.builtin.uri` or `ansible.legacy.uri` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-client.yml:3 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-client.yml:7 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-client.yml:12 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (get_url). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-client.yml:17 Use `ansible.builtin.get_url` or `ansible.legacy.get_url` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:2 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:6 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:11 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:16 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (set_fact). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:21 Use `ansible.builtin.set_fact` or `ansible.legacy.set_fact` instead.

fqcn: Use FQCN for builtin module actions (group). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:26 Use `ansible.builtin.group` or `ansible.legacy.group` instead.

fqcn: Use FQCN for builtin module actions (user). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:31 Use `ansible.builtin.user` or `ansible.legacy.user` instead.

fqcn: Use FQCN for builtin module actions (file). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:38 Use `ansible.builtin.file` or `ansible.legacy.file` instead.

fqcn: Use FQCN for builtin module actions (get_url). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:48 Use `ansible.builtin.get_url` or `ansible.legacy.get_url` instead.

fqcn: Use FQCN for builtin module actions (template). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:62 Use `ansible.builtin.template` or `ansible.legacy.template` instead.

fqcn: Use FQCN for builtin module actions (template). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:71 Use `ansible.builtin.template` or `ansible.legacy.template` instead.

fqcn: Use FQCN for builtin module actions (template). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:83 Use `ansible.builtin.template` or `ansible.legacy.template` instead.

fqcn: Use FQCN for builtin module actions (file). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:93 Use `ansible.builtin.file` or `ansible.legacy.file` instead.

fqcn: Use FQCN for builtin module actions (service). (fqcn[action-core])
~/.ansible/roles/minio/tasks/install-server.yml:102 Use `ansible.builtin.service` or `ansible.legacy.service` instead.

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - fqcn[action-core]  # Use FQCN for builtin actions.
  - yaml[truthy]  # Violations reported by yamllint.

                 Rule Violation Summary                  
 count tag               profile    rule associated tags 
     1 jinja[spacing]    basic      formatting (warning) 
     1 yaml[truthy]      basic      formatting, yaml     
     2 name[casing]      moderate   idiom (warning)      
    21 fqcn[action-core] production formatting           

Finished with 22 failure(s), 3 warning(s) on 11 files. Code passed min profile. No rating.
```

This is a sample case with a single role. My team uses a lot more third-party roles and collections where we cannot force maintainers to comply with ansible-lint rules.

_Originally posted by @btravouillon in https://github.com/ansible/ansible-lint/discussions/2067#discussioncomment-3775447_

**Comment by ssbarnea** [comment]
Create an ansible limt config file or be sure your code is git cloned. We use one of these to determine project root. Run with -vv and read the output, to see what it does.

**Comment by ssbarnea** [comment]
To be clear: cwd or argument path is not enough to determine project root.

**Comment by ssbarnea** [comment]
To be clear, I do expect that this is a corner-case bug, because I do not expect people to want to be notified about linting errors from their installed dependencies, basically we expect to never see errors reported from `~/.ansible/...`. Still, this one seems to affect very few people and I suspect is related to the way the tool is used. You must be holding it wrong! ;)

**Comment by btravouillon** [comment]
Thank you for looking at this.

> Create an ansible limt config file or be sure your code is git cloned

The code is not git cloned in this reproducer, but in the real world, it is.

Here is a new Dockerfile that will add the files to a git project `projectA`, then will clone this projet to `git-clone`. It also creates an empty `.ansible-lint` (I don't want to diverge from the default configuration).

_Note: I can reproduce on my dev workstation (not Docker nor ubi8/rhel8). The use of Docker here is for the purpose of reproducibility._

```
FROM registry.access.redhat.com/ubi8

RUN dnf install python39 git -y

RUN pip3 install -U pip
RUN pip install ansible ansible-lint==6.8.0b1

RUN git config --global user.email "fake@example.com"
RUN git config --global user.name "Fake"
RUN git init projectA
WORKDIR projectA/

RUN echo "roles:" > requirements.yml && \
    echo "  - name: minio" >> requirements.yml && \
    echo "    src: https://github.com/minio/ansible-minio" >> requirements.yml

RUN ansible-galaxy role install -r requirements.yml

RUN mkdir playbooks/
RUN echo "- name: Reproducer" > playbooks/minio.yml && \
    echo "  hosts: localhost" >> playbooks/minio.yml && \
    echo "  roles:" >> playbooks/minio.yml && \
    echo "    - role: minio" >> playbooks/minio.yml

# Empty config file since I don't have any default value to change
RUN touch .ansible-lint

RUN git add *
RUN git commit -m "Initial commit"
RUN git clone . /git-clone
WORKDIR /git-clone
```

```
docker build -t "ansible-lint:2067" .
docker run --rm -ti ansible-lint:2067
```

> Run with -vv and read the output, to see what it does.

I will.

> this one seems to affect very few people

In my production pipelines, I added `--exclude ~/.ansible/`. This workaround is easy to implement, thus it is highly possible more people than you think could be affected by this.

> You must be holding it wrong! ;)

I could agree with such statement, however I believe I use pretty standard ansible and ansible-lint configuration. The simplicity of the reproducer should highlight this (no custom `ANSIBLE_ROLE_PATHS`, default ansible configs, default ansible-lint configs, no environment variable that could break things). This does not differ that much from the documentation.

- https://ansible-lint.readthedocs.io/installing/#using-pip-or-pipx
- https://ansible-lint.readthedocs.io/usage/#linting-playbooks-and-roles

I do agree our views might differ on what is right and what is wrong. :wink:

**Comment by btravouillon** [comment]
Breakdown of the output of `ansible-lint -vv playbooks/`.

## Init
```
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/root/.cache/ansible-compat/0c4fa0', colored=True, configured=True, cwd=PosixPath('/git-clone'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], format='rich', lintables=['playbooks/'], listrules=False, listtags=False, write_list=None, parseable=False, quiet=0, rulesdirs=['/usr/local/lib/python3.9/site-packages/ansiblelint/rules'], skip_list=[], tags=[], verbosity=2, warn_list=['avoid-implicit', 'experimental', 'fqcn[action]', 'fqcn[redirect]', 'jinja[spacing]', 'name[casing]', 'name[play]', 'role-name', 'warning[empty-playbook]'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.{yaml,yml}'}, {'requirements': '**/meta/requirements.{yaml,yml}'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.{yaml,yml}'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, var_naming_pattern=None, offline=None, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, strict=False, rules={}, profile=None, progressive=False, rulesdir=[], use_default_rules=False, config_file=None, version=False, cache_dir_lock=<filelock._unix.UnixFileLock object at 0x7f82cc0a0280>)
DEBUG    /git-clone
INFO     Running ansible-galaxy role install -vr requirements.yml --roles-path /root/.cache/ansible-compat/0c4fa0/roles
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/0c4fa0/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/0c4fa0/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/0c4fa0/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/0c4fa0/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/0c4fa0/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/0c4fa0/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
```

## yamllint rules
```
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
```

## Discovered files
```
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z
```

Output of both commands.
```
[root@c8ef66881f64 git-clone]# git ls-files --cached --others --exclude-standard -z
playbooks/minio.ymlrequirements.yml[root@c8ef66881f64 git-clone]# 
[root@c8ef66881f64 git-clone]# git ls-files --deleted -z
[root@c8ef66881f64 git-clone]# 
```

## Syntax check
```
INFO     Executing syntax check on playbooks/minio.yml (0.64s)
DEBUG    Examining ~/.ansible/roles/minio/handlers/main.yml of type handlers
DEBUG    NAME 'minio_connection_type' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'minio_connection_type' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'minio_hostname' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'minio_hostname' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'minio_server_port' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'minio_server_port' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    Examining playbooks/minio.yml of type playbook
DEBUG    Examining ~/.ansible/roles/minio/tasks/main.yml of type tasks
DEBUG    Examining playbooks of type role
DEBUG    Examining ~/.ansible/roles/minio/vars/RedHat.yml of type vars
DEBUG    Examining ~/.ansible/roles/minio/vars/main.yml of type vars
[...very verbose, similar to previous lines...]
DEBUG    NAME '_minio_client_download_url' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME '_minio_client_download_url' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
WARNING  Listing 29 violation(s) that are fatal
jinja: Jinja2 spacing could be improved: {{ minio_server_ip }}:{{ minio_server_port}} -> {{ minio_server_ip }}:{{ minio_server_port }} (jinja[spacing])
~/.ansible/roles/minio/defaults/main.yml:30 Jinja2 template rewrite recommendation: `{{ minio_server_ip }}:{{ minio_server_port }}`.
[...]
name: All tasks should be named. (name[missing])
~/.ansible/roles/minio/tasks/main.yml:6 Task/Handler: include_tasks install-client.yml

DEBUG    Attempting to release lock 140199745684096 on /root/.cache/ansible-compat/0c4fa0/.lock
DEBUG    Lock 140199745684096 released on /root/.cache/ansible-compat/0c4fa0/.lock
You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - fqcn[action-core]  # Use FQCN for builtin actions.
  - name[missing]  # Rule for checking task and play names.
  - yaml[truthy]  # Violations reported by yamllint.
```

## Determined rule-profile order
```
DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-command-syntax': (7, 'basic'), 'deprecated-local-action': (8, 'basic'), 'deprecated-module': (9, 'basic'), 'inline-env-var': (10, 'basic'), 'key-order': (11, 'basic'), 'literal-compare': (12, 'basic'), 'jinja': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name[template]': (23, 'moderate'), 'name[imperative]': (24, 'moderate'), 'name[casing]': (25, 'moderate'), 'no-shorthand': (26, 'moderate'), 'spell-var-name': (27, 'moderate'), 'avoid-implicit': (28, 'safety'), 'latest': (29, 'safety'), 'package-latest': (30, 'safety'), 'risky-file-permissions': (31, 'safety'), 'risky-octal': (32, 'safety'), 'risky-shell-pipe': (33, 'safety'), 'galaxy': (34, 'shared'), 'ignore-errors': (35, 'shared'), 'layout': (36, 'shared'), 'meta-incorrect': (37, 'shared'), 'meta-no-info': (38, 'shared'), 'meta-no-tags': (39, 'shared'), 'meta-video-links': (40, 'shared'), 'meta-version': (41, 'shared'), 'meta-unsupported-ansible': (42, 'shared'), 'no-changed-when': (43, 'shared'), 'no-changelog': (44, 'shared'), 'no-handler': (45, 'shared'), 'no-relative-paths': (46, 'shared'), 'max-block-depth': (47, 'shared'), 'max-tasks': (48, 'shared'), 'unsafe-loop': (49, 'shared'), 'avoid-dot-notation': (50, 'production'), 'disallowed-ignore': (51, 'production'), 'fqcn': (52, 'production'), 'import-task-no-when': (53, 'production'), 'meta-no-dependencies': (54, 'production'), 'single-entry-point': (55, 'production'), 'use-loop': (56, 'production')}
```

## Summary
```
                 Rule Violation Summary                  
 count tag               profile    rule associated tags 
     1 jinja[spacing]    basic      formatting (warning) 
     2 name[missing]     basic      idiom                
     1 yaml[truthy]      basic      formatting, yaml     
     2 name[casing]      moderate   idiom (warning)      
    23 fqcn[action-core] production formatting           

Finished with 26 failure(s), 3 warning(s) on 11 files. Code passed min profile. No rating.
```

**Comment by btravouillon** [comment]
Adding some debug to

https://github.com/ansible/ansible-lint/blob/b4fd3ea0d72bedba59a7e829797b314fd421072a/src/ansiblelint/runner.py#L153-L160

```python
       # -- phase 2 ---
        if not matches:

            _logger.debug(
                "BT Lintables (2) %s of type %s",
                self.lintables,
                type(self.lintables),
            )

            # do our processing only when ansible syntax check passed in order
            # to avoid causing runtime exceptions. Our processing is not as
            # resilient to be able process garbage.
            matches.extend(self._emit_matches(files))

            _logger.debug(
                "BT Lintables (3) %s of type %s",
                self.lintables,
                type(self.lintables),
            )
```

The content of `self.lintables` changes (as well as the list of `files`).

```
INFO     Executing syntax check on playbooks/minio.yml (0.65s)
DEBUG    BT Lintables (2) {playbooks (role), playbooks/minio.yml (playbook)} of type <class 'set'>
DEBUG    BT Lintables (3) {~/.ansible/roles/minio/meta/main.yml (meta), ~/.ansible/roles/minio/handlers/main.yml (handlers), ~/.ansible/roles/minio/vars/RedHat.yml (vars), ~/.ansible/roles/minio/tasks/install-client.yml (tasks), ~/.ansible/roles/minio/tasks/main.yml (tasks), ~/.ansible/roles/minio/vars/Debian.yml (vars), ~/.ansible/roles/minio/tasks/install-server.yml (tasks), ~/.ansible/roles/minio/vars/main.yml (vars), ~/.ansible/roles/minio/defaults/main.yml (vars), playbooks (role), playbooks/minio.yml (playbook)} of type <class 'set'>
```

Looking at `_emit_matches()`, I can't say if the files should be excluded by `is_excluded` because they are part of `ANSIBLE_ROLES_PATH` or if they should not match in `ansiblelint.utils.find_children(lintable)`. (or maybe I totally misunderstand what is going on there :stuck_out_tongue: )

https://github.com/ansible/ansible-lint/blob/b4fd3ea0d72bedba59a7e829797b314fd421072a/src/ansiblelint/runner.py#L191-L208

**Comment by ssbarnea** [comment]
I already have a local fix, but I need to polish the tests, as the fix had some unexpected side-effects.

You are right, is_excluded() was flawed and did not exclude files outside the project_dir.

**Comment by jseiser** [comment]
We are hitting this with roles NOT in `~/.ansible`

```
❯ cat .ansible-lint
skip_list:
  - var-naming
  - braces
  - yaml
  - no-changed-when

exclude_paths:
  - .venv/ # implicit unless exclude_paths is defined in config
  - automation/
  - bin/
  - deployment/
  - .ansible-lint
  - .pre-commit-config.yaml
  - ../aws_development/
  - ../AWS_Development/
  - AWS_Development/
  - .gitlab-ci.yml

offline: true
```

```
jinja: Jinja2 spacing could be improved: /home/{{automation_user}}/.ssh/authorized_keys -> /home/{{ automation_user }}/.ssh/authorized_keys (jinja[spacing])
../aws_development/Automation/Ansible/roles/Set_authorized_keys/tasks/main.yml:75 Jinja2 template rewrite recommendation: `/home/{{ automation_user }}/.ssh/authorized_keys`.
```
